File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_10/problem_10.py
def sum_of_Factorials(n, s, i):
    if i == n:
        return 1 / s
    return 1 / s + sum_of_Factorials(n, s * (i + 1), i + 1)


def main():
    n = 5
    s = 1
    i = 1

    print(f"1! +...+ {n}! = {sum_of_Factorials(n, s, i):.3f}")

if __name__ == "__main__":
    main()

File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem10/Problem10.java
package recursive.problem10;

public class Problem10 {
    
    public static double sumOfFactorials(int n, double s, int i){
        if (i == n) {
            return (double)1 / s;
        }
        return (double)1 / s + sumOfFactorials(n, (double)s * (i + 1), i + 1);
    }

    public static void main(String[] args) {
        int n = 5;
        double sum = 1;
        int i = 1;

        System.out.printf("(1 / 1)! +...+ (1 / %d)! = %.3f", n, sumOfFactorials(n, sum, i));
    }
}
